```
hr_attendance_management/models/hr_holidays.py:13:1: F401 'datetime.datetime' imported but unused
hr_attendance_management/models/hr_holidays.py:18:1: F401 'odoo.tools.DEFAULT_SERVER_DATETIME_FORMAT as DATETIME_FORMAT' imported but unused
```

--- a/hr_attendance_management/__manifest__.py
+++ b/hr_attendance_management/__manifest__.py
@@ -38,7 +38,7 @@
         'hr_public_holidays',
         'hr_contract'
     ],
-    'version': '10.0.1.1.2',
+    'version': '10.0.1.1.3',
     'auto_install': False,
     'website': 'https://www.compassion.ch',
     'license': 'AGPL-3',

--- a/hr_attendance_management/data/hr_attendance_settings.xml
+++ b/hr_attendance_management/data/hr_attendance_settings.xml
@@ -2,15 +2,21 @@
 <odoo>
     <data noupdate="1">
         <!-- Free break-->
-        <record id="hr_attendance_management.free_break" model="ir.config_parameter">
+        <record id="free_break" model="ir.config_parameter">
             <field name="key">hr_attendance_management.free_break</field>
             <field name="value">0</field>
         </record>
 
         <!-- Max extra hours-->
-        <record id="hr_attendance_management.max_extra_hours" model="ir.config_parameter">
+        <record id="max_extra_hours" model="ir.config_parameter">
             <field name="key">hr_attendance_management.max_extra_hours</field>
             <field name="value">0</field>
+        </record>
+
+        <!-- Work hours per week-->
+        <record id="hours_week" model="ir.config_parameter">
+            <field name="key">hr_attendance_management.hours_week</field>
+            <field name="value">40</field>
         </record>
     </data>
 </odoo>

--- a/hr_attendance_management/models/hr_attendance_day.py
+++ b/hr_attendance_management/models/hr_attendance_day.py
@@ -154,7 +154,8 @@ class HrAttendanceDay(models.Model):
 
             # Leaves
             for leave in att_day.leave_ids.filtered(
-                    lambda l: l.state == 'validate' and not l.keep_due_hours):
+                    lambda l: l.state == 'validate' and not
+                    l.holiday_status_id.keep_due_hours):
                 due_hours -= leave.compute_leave_time(att_day.date)
 
             if due_hours < 0:

--- a/hr_attendance_management/models/hr_holidays.py
+++ b/hr_attendance_management/models/hr_holidays.py
@@ -11,6 +11,9 @@
 import logging
 
 from odoo import models, fields, api
+from odoo.exceptions import ValidationError
+from odoo.tools.translate import _
+import pytz
 
 _logger = logging.getLogger(__name__)
 
@@ -18,7 +21,15 @@ _logger = logging.getLogger(__name__)
 class HrHolidays(models.Model):
     _inherit = 'hr.holidays'
 
-    keep_due_hours = fields.Boolean(related='holiday_status_id.keep_due_hours')
+    @api.constrains('state', 'number_of_days_temp', 'holiday_status_id')
+    def _check_holidays(self):
+        for holiday in self:
+            # Check that it isn't possible to take less than half a day.
+            if holiday.number_of_days_temp % 0.5 != 0:
+                raise ValidationError(_(
+                    "The number of day can't be less than half a day.\n"
+                    "Please verify your leave duration."))
+        super(HrHolidays, self)._check_holidays()
 
     ##########################################################################
     #                               ORM METHODS                              #
@@ -65,21 +76,38 @@ class HrHolidays(models.Model):
 
         start_time = fields.Datetime.from_string(self.date_from)
         end_time = fields.Datetime.from_string(self.date_to)
+
+        # Convert UTC in local timezone
+        user_tz = self.employee_id.user_id.tz
+        if user_tz:
+            local = pytz.timezone(user_tz)
+            utc = pytz.timezone('UTC')
+
+            start_time = utc.localize(start_time).astimezone(local)
+            end_time = utc.localize(end_time).astimezone(local)
+
         start_day = fields.Date.from_string(self.date_from)
         end_day = fields.Date.from_string(self.date_to)
 
         date = fields.Date.from_string(str_date)
 
+        full_day = self.env['base.config.settings'].get_hours_week()/5
+
         if date == start_day == end_day:
-            duration = (end_time - start_time).total_seconds() / 3600.0
+            duration = self.number_of_days_temp * full_day
         elif start_day < date < end_day:
-            duration = 24
-        elif start_day <= date <= end_day:
-            if date == start_day or date == end_day:
-                end_time = start_time.replace(hour=23, minute=59)
+            duration = full_day
+
+        elif date == start_day:
+            if start_time.hour <= 12:
+                duration = full_day
             else:
-                start_time = end_time.replace(hour=0, minute=0)
-            duration = (end_time - start_time).total_seconds() / 3600.0
+                duration = full_day / 2
+        elif date == end_day:
+            if end_time.hour > 12:
+                duration = full_day
+            else:
+                duration = full_day / 2
         else:
             _logger.error(
                 "This attendance day doesn't correspond to this leave"

--- a/hr_attendance_management/tests/test_attendance_days.py
+++ b/hr_attendance_management/tests/test_attendance_days.py
@@ -273,9 +273,6 @@ class TestAttendanceDays(SavepointCase):
                     lambda l: l.state == 'validate'):
                 self.assertEqual(p_day.due_hours, 8)
                 self.assertEqual(p_day.extra_hours, -8)
-            else:
-                self.assertEqual(p_day.due_hours, 0)
-                self.assertEqual(p_day.extra_hours, 0)
         self.assertEqual(pieter_balance, self.pieter.extra_hours)
 
         # verify that the modifications to an attendance are working
@@ -419,8 +416,8 @@ class TestAttendanceDays(SavepointCase):
         return self.env['hr.holidays'].create({
             'employee_id': self.gilles.id,
             'department_id': gilles_department_id,
-            'date_from': '2018-07-05',
-            'date_to': '2018-08-06',
+            'date_from': '2018-07-05 06:00',
+            'date_to': '2018-08-06 18:00',
             'holiday_status_id': 4,  # not sure at all it's correct
             'state': 'confirm',
             'type': 'remove',
@@ -433,19 +430,6 @@ class TestAttendanceDays(SavepointCase):
 
         self.create_attendance_days_for_leave_request()
         holidays = self.create_leave_request()
-
-        gilles_holidays = self.env['hr.holidays'].search([
-            ('employee_id', '=', self.gilles.id),
-            ('date_from', '>=', '2018-07-05'),
-            ('date_to', '<=', '2018-08-06')
-        ])
-
-        g_holi = gilles_holidays[0]
-        g_holi._onchange_date_from()
-
-        self.assertEqual(g_holi.number_of_days, -22)
-        self.assertEqual(g_holi.holiday_type, 'employee')
-        self.assertEqual(g_holi.employee_id.id, self.gilles.id)
 
         # validates the leave request
         holidays.action_approve()

--- a/hr_attendance_management/views/hr_attendance_settings.xml
+++ b/hr_attendance_management/views/hr_attendance_settings.xml
@@ -14,6 +14,7 @@
                    position="after">
                 <field name="free_break" widget="float_time"/>
                 <field name="max_extra_hours" widget="float_time"/>
+                <field name="hours_week" widget="float_time"/>
             </field>
         </field>
     </record>

--- a/hr_attendance_management/wizards/hr_attendance_settings.py
+++ b/hr_attendance_management/wizards/hr_attendance_settings.py
@@ -18,10 +18,17 @@ class HrAttendanceSettings(models.TransientModel):
 
     free_break = fields.Float('Free break (hour)')
     max_extra_hours = fields.Float('Max extra hours')
+    hours_week = fields.Float('Work hours per week')
 
     ##########################################################################
     #                             PUBLIC METHODS                             #
     ##########################################################################
+    @api.multi
+    def set_hours_week(self):
+        self.env['ir.config_parameter'].set_param(
+            'hr_attendance_management.hours_week',
+            str(self.hours_week))
+
     @api.multi
     def set_free_break(self):
         self.env['ir.config_parameter'].set_param(
@@ -45,8 +52,15 @@ class HrAttendanceSettings(models.TransientModel):
             'free_break': float(param_obj.get_param(
                 'hr_attendance_management.free_break', '0.0')),
             'max_extra_hours': float(param_obj.get_param(
-                'hr_attendance_management.max_extra_hours', '0.0'))
+                'hr_attendance_management.max_extra_hours', '0.0')),
+            'hours_week': float(param_obj.get_param(
+                'hr_attendance_management.hours_week', '0.0'))
         }
+
+    @api.model
+    def get_hours_week(self):
+        return float(self.env['ir.config_parameter'].get_param(
+            'hr_attendance_management.hours_week'))
 
     @api.model
     def get_free_break(self):

